deepmd/dpmodel/model/linear_atomic_model.py

```diff
@@ -0,0 +1,288 @@
+        )
+        return [p[0] for p in zipped], [p[1] for p in zipped]
+
+    def forward_atomic(
+        self,
+        extended_coord,
+        extended_atype,
+        nlist,
+        mapping: Optional[np.ndarray] = None,
+        fparam: Optional[np.ndarray] = None,
+        aparam: Optional[np.ndarray] = None,
+    ) -> Dict[str, np.ndarray]:
+        """Return atomic prediction.
+
+        Parameters
+        ----------
+        extended_coord
+            coodinates in extended region, (nframes, nall * 3)
+        extended_atype
+            atomic type in extended region, (nframes, nall)
+        nlist
+            neighbor list, (nframes, nloc, nsel).
+        mapping
+            mapps the extended indices to local indices.
+        fparam
+            frame parameter. (nframes, ndf)
+        aparam
+            atomic parameter. (nframes, nloc, nda)
+
+        Returns
+        -------
+        result_dict
+            the result dict, defined by the fitting net output def.
+        """
+        nframes, nloc, nnei = nlist.shape
+        self.extended_coord = extended_coord.reshape(nframes, -1, 3)
+        sorted_rcuts, sorted_sels = self._sort_rcuts_sels()
+        nlists = build_multiple_neighbor_list(
+            self.extended_coord,
+            nlist,
+            sorted_rcuts,
+            sorted_sels,
+        )
+        raw_nlists = [
+            nlists[get_multiple_nlist_key(rcut, sel)]
+            for rcut, sel in zip(self.get_rcuts(), self.get_sel())
+        ]
+        self.nlists_ = [
+            nl if not dt else nlist_distinguish_types(nl, extended_atype, sel)
+            for dt, nl, sel in zip(
+                self.distinguish_type_list, raw_nlists, self.get_original_sels()
+            )
+        ]
+        ener_list = [
+            model.forward_atomic(
+                self.extended_coord,
+                extended_atype,
+                nl,
+                mapping,
+            )["energy"]
+            for model, nl in zip(self.models, self.nlists_)
+        ]
+        self.weights = self._compute_weight()
+        self.atomic_bias = None
+        if self.atomic_bias is not None:
+            raise NotImplementedError("Need to add bias in a future PR.")
+        else:
+            fit_ret = {
+                "energy": np.sum(np.stack(ener_list) * np.stack(self.weights), axis=0),
+            }  # (nframes, nloc, 1)
+        return fit_ret
+
+    def fitting_output_def(self) -> FittingOutputDef:
+        return FittingOutputDef(
+            [
+                OutputVariableDef(
+                    name="energy", shape=[1], reduciable=True, differentiable=True
+                )
+            ]
+        )
+
+    def serialize(self) -> dict:
+        return {
+            "models": [model.serialize() for model in self.models],
+        }
+
+    @classmethod
+    def deserialize(cls, data) -> "LinearAtomicModel":
+        models = [DPAtomicModel.deserialize(model) for model in data["models"]]
```

Comment:
it seems that the `deserialization` would not work if one model is `DPAtomicModel` and another is `PairTabModel` 

please refer to the serialize and deserialize methods used in below 
https://github.com/deepmodeling/deepmd-kit/blob/f1baf198b2c703637215942428d2ef24575b0e92/deepmd/dpmodel/model/dp_atomic_model.py#L122-L141